homeassistant/components/fitbit/sensor.py

```diff
@@ -360,6 +364,11 @@ def __init__(
         self.extra = extra
         if self.extra is not None:
             self._attr_name = f"{self.extra.get('deviceVersion')} Battery"
+            self._attr_unique_id = f"fitbit_{user_profile['encodedId']}_{self.extra.get('deviceVersion')}_battery"
+        else:
+            self._attr_unique_id = (
+                f"fitbit_{user_profile['encodedId']}_{description.key}"
+            )
```

Comment:
Please remove "fitbit_" prefix from the unique_id.

Reply:
Ahh sorry I missed that in the docs:

> Entities should not include the domain (e.g., your_integration) and platform type (e.g., light) in their Unique ID as the system already accounts for these identifiers.

This is all starting to make sense, thanks for your patience! I'll also add new commits and stop squashing and force pushing